lang_python.adoc

```diff
@@ -13,130 +13,171 @@ To generate Python modules, Kaitai Struct compiler should be invoked with
 `-t python` option:
 
 [source,shell]
-kaitai-struct-compiler -t python my_spec.ksy
+kaitai-struct-compiler -t python --python-package . my_spec.ksy
 
 This normally generates `my_spec.py` parser module, which is compatible
-with both Python 2 and 3.
+with both Python 2 and 3. It is strongly recommended to use Python 3
+because Python 2 is EOL since 2020-01-01. You can also install
+`python-is-python3` on Ubuntu focal to make `python3` be used via
+`python` command.
+
+`--python-package` optional CLI arg allows you to select a parent package
+name used by one generated file to import symbols from another one.
+
+If you **don't intend to embed** the KSC-generated files and just want to
+play a bit in IPython/Jupyter shell in the same dir, **remove** this arg,
+otherwise you get an `ImportError` .
+
+If you **intend to embed** the compilation result into an own package,
+it is recommended **to set the arg to `.`**, which means "current package",
+rather than to the real  name of the subpackage, because setting it
+to a real name will cause breakage when you (or anyone else) want
+to reorganize or rename your package. It will cause KSC to generate
```

Comment:
I would shorten this paragraph to only talk about `--python-package .` and leave out the discussion about `--python-package absolute.package.name`. That way it's less confusing for people who don't already know what `--python-package´ does.

I don't know of any cases where `--python-package .` does not work and you have to use `--python-package absolute.package.name` instead. I would either remove the discussion of the absolute name variant completely, or move it somewhere to the bottom.

Reply:
The problem that in PEP8 it is written it is advised to use the absolute name. After it it is written that relative imports are also allowed. But some people are PEP8 fanatics ...

Reply:
It's only a recommendation, so it probably won't cause problems with automated code style checkers (which shouldn't be applied to generated code anyway). Besides, the Python code generated by KS already violates other PEP 8 guidelines, such as those about how to group imports, so I think nobody expects KS to generate strictly PEP 8-compliant Python code.

In any case, I still think it's not important enough to mention this early in the section.

Reply:
Shortened the section and removed all the unneeded details.